test/tests_for_sdkmr/EE2Runjob_test.py

```diff
@@ -0,0 +1,268 @@
+"""
+
+# Incomplete by a long way. Will add more unit tests as they come up.
+
+from bson.objectid import ObjectId
+from logging import Logger
+from unittest.mock import create_autospec, call
+from execution_engine2.db.models.models import Job, JobInput, JobRequirements, Meta
+from execution_engine2.sdk.EE2Runjob import EE2RunJob, JobPermissions
+from execution_engine2.sdk.SDKMethodRunner import SDKMethodRunner
+from execution_engine2.utils.CatalogUtils import CatalogUtils
+from execution_engine2.utils.Condor import (
+    Condor,
+    CondorResources,
+    SubmissionInfo,
+)
+from execution_engine2.utils.KafkaUtils import (
+    KafkaClient,
+    KafkaQueueChange,
+    KafkaCreateJob,
+)
+from execution_engine2.utils.SlackUtils import SlackClient
+from lib.execution_engine2.db.MongoUtil import MongoUtil
+from installed_clients.WorkspaceClient import Workspace
+from installed_clients.CatalogClient import Catalog
+
+
+def test_run_as_admin():
+    """
+    A basic unit test of the run() method with an administrative user.
+
+    This test is a fairly minimal test of the run() method. It does not exercise all the
+    potential code paths or provide all the possible run inputs, such as job parameters, cell
+    metadata, etc.
+    """
+    # The amount of mocking required here implies the method should be broken up into smaller
+    # classes that are individually mockable. Or maybe it's just really complicated and this
+    # is the best we can do. Worth looking into at some point though.
+
+    # We intentionally do not check the logger methods as there are a lot of them and this is
+    # already a very large test. This may be something to be added later when needed.
+    sdkmr = create_autospec(SDKMethodRunner, spec_set=True, instance=True)
+    logger = create_autospec(Logger, spec_set=True, instance=True)
+    ws = create_autospec(Workspace, spec_set=True, instance=True)
+    catutils = create_autospec(CatalogUtils, spec_set=True, instance=True)
+    catalog = create_autospec(Catalog, spec_set=True, instance=True)
+    condor = create_autospec(Condor, spec_set=True, instance=True)
+    kafka = create_autospec(KafkaClient, spec_set=True, instance=True)
+    mongo = create_autospec(MongoUtil, spec_set=True, instance=True)
+    slack = create_autospec(SlackClient, spec_set=True, instance=True)
+    sdkmr.get_logger.return_value = logger
+    sdkmr.get_workspace.return_value = ws
+    sdkmr.get_catalog_utils.return_value = catutils
+    catutils.get_catalog.return_value = catalog
+    sdkmr.get_condor.return_value = condor
+    sdkmr.get_kafka_client.return_value = kafka
+    sdkmr.get_slack_client.return_value = slack
+    sdkmr.get_mongo_util.return_value = mongo
+    sdkmr.get_user_id.return_value = "someuser"
+    sdkmr.get_token.return_value = "tokentokentoken"
+    sdkmr.check_as_admin.return_value = True
+    sdkmr.save_job.return_value = "603051cfaf2e3401b0500982"
+    ws.get_object_info3.return_value = {"paths": [["1/2/3"], ["4/5/6"]]}
```

Comment:
can you alphabetise these so it's easy for people to find whatever method they're looking for?

Reply:
They're actually in the same order that the calls occur in the code, and I'd like to keep them that way unless there's a good reason not to. I'll add comments to that effect.